data/filter/Utils.ts

```diff
@@ -146,3 +148,51 @@ export function combineValueFilters(filters = []): Filter[] {
             : filters;
     });
 }
+
+/**
+ * Serializes a filter by storing additional type information needed for Date and LocalDate values.
+ */
+export function serializeFilter(filter: FilterLike): string {
+    if (isNil(filter)) return filter;
+    if (isArray(filter)) {
+        return JSON.stringify(filter.map(filter => JSON.parse(serializeFilter(filter))));
+    } else if ('filters' in filter) {
+        return JSON.stringify({
+            ...filter,
+            filters: filter.filters.map(filter => JSON.parse(serializeFilter(filter)))
+        });
+    } else if ('value' in filter) {
+        return JSON.stringify({...filter, type: filter.value?.constructor.name});
+    } else {
+        throw XH.exception(`Unable to serialize filter ${filter}`);
+    }
+}
+
+/**
+ * Deserializes a filter, respecting the value type information stored by `serializeFilter` for
+ * Date and LocalDate values. Note that this method will throw an exception if it encounters
+ * a filter it can in no way deserialize; however, the absence of an exception does not guarantee
+ * that the filter is valid.
+ */
+export function deserializeFilter(filter: string | PlainObject): FilterChooserFilterLike {
```

Comment:
Not sure about use of `FilterChooserFilterLike` here - or at least not sure if that's the name we want to be seeing here.  Is it too closely bound to that one component / implementation, or is it the thing we already have that nicely captures the compound+field filter only subset of filters.

Also, looking at that type itself, should we swap out `FilterChooserFilter` (the two concreete types) for `Filter`?  And should we consider a rename, or leave it be?

Finally, should the `serialize` method also take the same, given that it will throw if passed a function filter or spec? Wanted to talk through the relationship between say the type signature there and the final throw

Reply:
I agree the name isn't right here, but I went with it since it's the union type we have for serializable filters.  Maybe another type alias or a rename would make sense?

I'm not sure I quite follow the 2nd question.

I do think serialize should take the same. The tricky part is that our `CompoundFilter.filters` is typed as `Filter[]`, so when we recursively call `serializeFilter` with those nested filters, we'd get a TS error.  I suppose we could filter the `filters` first to make that type-safe. That probably is the cleanest approach and more clearly communicates the allowed types.